storage/remote/queue_manager.go

```diff
@@ -158,6 +168,13 @@ func NewStorageQueueManager(tsdb StorageClient, cfg *StorageQueueManagerConfig)
 // sample on the floor if the queue is full.
 // Always returns nil.
 func (t *StorageQueueManager) Append(s *model.Sample) error {
+	s.Metric = model.Metric(
```

Comment:
I could move the external labels in here too, even though they are the same for all queues. Otherwise I'd have to move this queue-specific relabeling into `remote.Storage` as well, but it belongs into each queue?